vertx-web-client/src/main/java/io/vertx/ext/web/client/HttpRequest.java

```diff
@@ -242,4 +242,11 @@
    */
   void send(Handler<AsyncResult<HttpResponse<T>>> handler);
 
+  HttpMethod getMethod();
```

Comment:
for now I would prefer to have these internal and have the interceptor to make a cast to the impl